src/compute/src/sink/persist_sink.rs

```diff
@@ -845,6 +845,24 @@ where
             correction_oks.advance_by(&persist_frontier);
             correction_errs.advance_by(&persist_frontier);
 
+            // Discard batch descriptions whose upper is already not beyond the
+            // persist frontier. Those have no chance of being applied to the
+            // shard succesfully.
+            let persist_upper = write.shared_upper();
+            in_flight_batches
+                .retain(|(_lower, upper), _cap| PartialOrder::less_equal(&persist_upper, upper));
```

Comment:
This is not incorrect, but a batch can already not be appended when its lower is less than the persist frontier. Is there a reason to compare with the batch upper instead?

Reply:
I don't think there's a good reason. I'll change this and the other place to use the lower 👌